spec/javascripts/components/table-filter-spec.js

```diff
@@ -0,0 +1,121 @@
+        '</table>' +
+        '<p class="js-gem-c-table__message govuk-!-display-none">That search returns no results.</p>' +
+      '</div>'
+
+    window.setFixtures(FIXTURE)
+  })
+
+  describe('When the component is intialised', function () {
+    beforeEach(function () {
+      element = document.querySelector('[data-module="table-filter"]')
+      filter = document.querySelector('.js-gem-c-table__filter')
+      searchInput = document.querySelector('input')
+      tableFilter = new GOVUK.Modules.TableFilter(element)
+      event = new window.Event('input')
+      hiddenClass = tableFilter.hiddenClass
+
+      spyOn(tableFilter, 'updateRows')
+      tableFilter.init()
+    })
+
+    it('Displays the filter section', function () {
+      expect(filter.classList.contains(hiddenClass)).toBe(false)
+    })
+
+    it('Sets up the event listener', function () {
+      searchInput.dispatchEvent(event)
+      expect(tableFilter.updateRows).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  describe('When the user inputs a value', function () {
+    beforeEach(function () {
+      element = document.querySelector('[data-module="table-filter"]')
+      filterCount = document.querySelector('.js-filter-count')
+      searchInput = document.querySelector('input')
+      rows = document.querySelectorAll('.js-govuk-table__row')
+      message = document.querySelector('.js-gem-c-table__message')
+      tableFilter = new GOVUK.Modules.TableFilter(element)
+      hiddenClass = tableFilter.hiddenClass
+      event = new window.Event('input')
+
+      tableFilter.init()
+    })
+
+    it('Hides all rows and displays a message when a non-matching term is input', function () {
+      searchInput.value = 'a'
+      searchInput.dispatchEvent(event)
+      expect(rows[0].classList.contains(hiddenClass)).toBe(true)
+      expect(rows[1].classList.contains(hiddenClass)).toBe(true)
+      expect(message.classList.contains(hiddenClass)).toBe(false)
+      expect(filterCount.textContent).toEqual('Number of filtered rows: 0')
+    })
+
+    it('Shows the correct rows and does not display a message when a matching term is input', function () {
+      searchInput.value = 'b'
+      searchInput.dispatchEvent(event)
+      expect(rows[0].classList.contains(hiddenClass)).toBe(false)
+      expect(rows[1].classList.contains(hiddenClass)).toBe(true)
+      expect(message.classList.contains(hiddenClass)).toBe(true)
+      expect(filterCount.textContent).toEqual('Number of filtered rows: 1')
+    })
+
+    it('Shows the correct rows when the case of the input term differs from the table row content', function () {
+      searchInput.value = 'OP'
+      searchInput.dispatchEvent(event)
+      expect(rows[0].classList.contains(hiddenClass)).toBe(true)
+      expect(rows[1].classList.contains(hiddenClass)).toBe(false)
+      expect(filterCount.textContent).toEqual('Number of filtered rows: 1')
+    })
+
+    it('Shows all the rows when the input is cleared', function () {
```

Comment:
This test is much better, but can you also check that the hidden number of rows text changes when the query is entered and then is reset when the query is cleared? I know that other tests check that it gets set correctly with matching and non-matching queries, but this is the only place where the test clears the input and we should check that everything returns to normal as expected.